### PR TITLE
job-builder: introduced secure base-job-image

### PR DESCRIPTION
**Pull request by mivanov1988**
# What
As part of our initiative to improve the overall security of the VDK project, we need to apply some general hardenings to the base job image.

# Why
Introduced base job image based on the lightweight [Photon OS](https://vmware.github.io/photon/assets/files/html/3.0/Introduction.html). The general hardenings will be applied in a separate PR.

# Testing done
Tested on a local Kind cluster

Signed-off-by: Miroslav Ivanov miroslavi@vmware.com

### CODE REVIEW

**Comment by mivanov1988** [comment]
> can we remove the insecure image? To keep the code clean? Is there are use case for the insecure one?

Some of our internal deployments depend on it.

**Comment by mivanov1988** [comment]
> Can they migrate to the secure ones soon though?

It is not planned yet.

File: projects/control-service/projects/job-base-image-secure/publish-job-base.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    image_repo="$VDK_DOCKER_REGISTRY_URL/$name"
+    image_tag="$image_repo:$VERSION_TAG"
+
+    docker build -t "$image_tag" -t "$image_repo:latest" -f "$SCRIPT_DIR/$docker_file" $arguments "$SCRIPT_DIR"
+    docker push "$image_tag"
+    docker push "$image_repo:latest"
+}
+
+build_and_push_image \
+    "data-job-base-python-3.10-secure" \
```
Comment (murphp15) [inline projects/control-service/projects/job-base-image-secure/publish-job-base.sh:24]: how do you ensure that it is python 3.10?

**Review by murphp15 — APPROVED** [review]

File: projects/control-service/projects/job-base-image-secure/publish-job-base.sh (hunk 1 above)
Comment (mivanov1988) [reply projects/control-service/projects/job-base-image-secure/publish-job-base.sh:24]: Good point, I will try to specify it in the image.

File: projects/control-service/projects/job-base-image-secure/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Job base image
+
+Job base image is the container "base" image used when building per data job custom image during deployment.
+
+This directory provides the source of some base images for standard python versions.
+It's used by default installation of VDK and should also serve as an example on how operators of VDK can build their own base image.
+
+The current base image installs supporting libraries for Oracle database
```
Comment (tozka) [inline projects/control-service/projects/job-base-image-secure/README.md:8]: The README looks wrong. Do we plan to support Oracle libraries in view that now there's pure python library that can be used without native client installations required (https://python-oracledb.readthedocs.io) ?

File: projects/control-service/cicd/.gitlab-ci.yml [hunk 3]
```diff
@@ -140,6 +140,26 @@ control_service_publish_job_base_image:
...
+    - apk add --no-cache bash
+    - docker login --username "${VDK_DOCKER_REGISTRY_USERNAME}" --password "${VDK_DOCKER_REGISTRY_PASSWORD}" "${VDK_DOCKER_REGISTRY_URL}"
+    - cd projects/control-service/projects/job-base-image-secure
+    - export VERSION_TAG="1.$CI_PIPELINE_ID"
+    - bash -ex ./publish-job-base.sh
+  retry: !reference [.control_service_retry, retry_options]
+  only:
+    refs:
+      - main
+      - external_pull_requests
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/cicd/.gitlab-ci.yml:156]: Yes, it is for testing purposes.

File: projects/control-service/projects/job-base-image-secure/README.md (hunk 2 above)
Comment (mivanov1988) [reply projects/control-service/projects/job-base-image-secure/README.md:8]: Correct, I will update it.

**Review by tozka — APPROVED** [review]
What are the benefits of photon vs python-slim (which is the default one) ? 

In any case looks good to me. Offering more secure option is certainly a good think .

**Comment by mivanov1988** [comment]
> 

One of the key features of interest of the Photon OS is: "The kernel and other aspects of the operating system are built with an emphasis on security.".

On scanning the Photon OS image, the vulnerabilities are as follows: `Tested 35 dependencies for known issues, no vulnerable paths found.`

On scanning python:3.10-slim the vulnerabilities are as follows: Tested 106 dependencies for known issues, found 47 issues.

**Comment by tozka** [comment]
> > 
> 
> One of the key features of interest of the Photon OS is: "The kernel and other aspects of the operating system are built with an emphasis on security.".
> 
> On scanning the Photon OS image, the vulnerabilities are as follows: `Tested 35 dependencies for known issues, no vulnerable paths found.`
> 
> On scanning python:3.10-slim the vulnerabilities are as follows: `Tested 106 dependencies for known issues, found 47 issues.`

Thanks. That make sense.

**Pull request merged**